pkg/nfd-master/nfd-master.go

```diff
@@ -758,21 +797,29 @@ func (m *nfdMaster) nfdAPIUpdateAllNodes() error {
 	}
 
 	for _, node := range nodes.Items {
-		m.nodeUpdaterPool.addNode(node.Name)
+		m.updaterPool.addNode(node.Name)
 	}
 
 	return nil
 }
 
-func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.Node) error {
-	if m.nfdController == nil || m.nfdController.featureLister == nil {
-		return nil
+// mergeNodeFeatures merges the features of the given NodeFeature objects into a single NodeFeatureSpec.
+func (m *nfdMaster) mergeNodeFeatures(nodeName string) (*nfdv1alpha1.NodeFeature, error) {
```

Comment:
It's so easy to nit about code written by others 🙄 But now reading this the function name is a bit misleading to me.

```suggestion
func (m *nfdMaster) getMergedNodeFeatures(nodeName string) (*nfdv1alpha1.NodeFeature, error) {
```

**OR**

```suggestion
func (m *nfdMaster) getAndMergeNodeFeatures(nodeName string) (*nfdv1alpha1.NodeFeature, error) {
```